web/pdf_page_view.js

```diff
@@ -1064,7 +1064,11 @@ class PDFPageView extends BasePDFPageView {
       if (this.annotationLayer) {
         await this.#renderAnnotationLayer();
 
-        if (this.#enableAutoLinking && this.annotationLayer) {
+        if (
+          this.#enableAutoLinking &&
+          this.annotationLayer &&
+          !pdfPage.isPureXfa
```

Comment:
Does `this.textLayer` in XFA documents exist? Because if we can replace this `!pdfPage.isPureXfa` with `!this.textLayer`, it would make it clearer to understand why `this.#injectLinkAnnotations` is gated by these conditions.

Reply:
It does not, see:
https://github.com/mozilla/pdf.js/blob/50c573d16d2dbd8c2bdb3bd4c6f5e85f46602c79/web/pdf_page_view.js#L939-L959